src/System.Net.Primitives/src/System/Net/CookieContainer.cs

```diff
@@ -1062,17 +1045,9 @@ public IEnumerator GetEnumerator()
             }
         }
 
-        public object SyncRoot
-        {
-            get
-            {
-                Debug.Assert(m_list != null, $"{nameof(PathList)} should never be default initialized and only ever created with {nameof(Create)}.");
-                return m_list;
-            }
-        }
+        public object SyncRoot => m_list;
```

Comment:
Since you're already changing this line, why not change it back to `m_list.SyncRoot` to match desktop?